docs/api/index.md

```diff
@@ -2964,6 +2992,25 @@ IntersectionObserver === undefined
   vi.runAllTimers()
   ```
 
+### vi.runAllTimersAsync
+
+- **Type:** `async() => Promise<Vitest>`
+
+  This method will asynchronously invoke every initiated timer until the timers queue is empty. It means that every timer called during `runAllTimersAsync` will be fired even asynchronous timers. If you have an infinite interval,
+  it will throw after 10 000 tries. For example this will log `2, 1`:
+
+  ```ts
+  setTimeout(async () => {
+    await setTimeout(() => {
```

Comment:
this has no effect, is it? (unless imported form `timers/promises`, which is not specified in this code block)

Reply:
what do you mean it has no effect ? the `await` ?

Reply:
Yes, the await

Reply:
it actually has an impact
I can't find much documentation on this but having a `Promise` in a `setTimeout` has a different behavior from having a synchronous callback

the impact on timers is that this code would not work for `runAllTimers`

Reply:
okay this was not clear to me at all

I've been playing with these a bit and my understanding is that it doesn't change anything in node timers, but it changes in sinonjs/fake-timers

for their synchronous api, an asynchronous callback in a `setTimeout` will block the timer

```typescript
vi.useFakeTimers()

setTimeout(async () => {
  console.log(1)
  await Promise.resolve(true) // this promise will block the timer
  
  console.log(2) // happens after termination
}, 100)

vi.runAllTimers()
```

so you are right, for node it makes no difference, but for our fake timers it does

I will update the example to better reflect on that